stl/inc/xstring

```diff
@@ -2459,23 +2490,116 @@ private:
             int>;
 #endif // _HAS_CXX17
 
-public:
-    _CONSTEXPR20 basic_string(const basic_string& _Right)
-        : _Mypair(_One_then_variadic_args_t{}, _Alty_traits::select_on_container_copy_construction(_Right._Getal())) {
-        auto&& _Alproxy = _GET_PROXY_ALLOCATOR(_Alty, _Getal());
-        _Container_proxy_ptr<_Alty> _Proxy(_Alproxy, _Mypair._Myval2);
-        _Construct_lv_contents(_Right);
-        _Proxy._Release();
-    }
+#if defined(__SANITIZE_ADDRESS__) || defined(_ANNOTATE_STRING)
+    class _NODISCARD _Asan_annotater {

```

Comment:
Is the name distinct from `vector` annotater?

Reply:
no it is the same name but both are classes inside `std::vector` and `std::string` repectively so I think they should be fine from an access / name mangling point of view

Reply:
Now that I see it, the class is non-constructible but marked `_NODISCARD`.

I mean it is not wrong but 🤔